utilities/tools/Get-ModulesFeatureOutline.ps1

```diff
@@ -0,0 +1,193 @@
+﻿<#
+.SYNOPSIS
+Get an outline of all modules features for each module contained in the given path
+
+.DESCRIPTION
+Get a list of objects that outline of all modules features for each module contained in the given path (for example child-modules, RBAC, Private Endpoints, etc.)
+
+NOTE: Currently only supports modules using the Bicep DSL
+
+.PARAMETER ModuleFolderPath
+Optional. The path to the modules.
+
+.PARAMETER ReturnMarkdown
+Optional. Instead of returning the list of objects, instead format them into a markdown table and return it as a string.
+
+.PARAMETER BreakMarkdownModuleNameAt
+Optional. When `ReturnMarkdown` is set to true you can use this number to control if & where you'd want to line break the ModuleName column. Defaults to 1 (i.e., right after the provider namepsace).
+
+.PARAMETER OnlyTopLevel
+Optional. Only consider top-level modules (that is, no child-modules).
+
+.EXAMPLE
+Get-ModulesFeatureOutline
+
+Get an outline of all modules in the default module path.
+
+.EXAMPLE
+Get-ModulesFeatureOutline -ReturnMarkdown -OnlyTopLevel
+
+Get an outline of top-level modules in the default module path, formatted in a markdown table.
+
+.EXAMPLE
+Get-ModulesFeatureOutline -ReturnMarkdown -BreakMarkdownModuleNameAt 2
+
+Get an outline of all modules in the default module path, formatted in a markdown table - with the module name column split after the top-level (i.e., <ProviderNamespace>/<ResourceType)
+
+.NOTES
+Children (if any) are displayed in format `[L1:5, L2:4, L3:1]`. Each item (separated via ',') shows the level of nesting in the front (e.g. L1) and the number of children in this level (separated by a colon ':').
+In the above example, the module has 5 direct children, 4 of them have direct children themselves and 1 of them has 1 more child.
+#>
+function Get-ModulesFeatureOutline {
+
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $false)]
+        [string] $ModuleFolderPath = (Join-Path (Split-Path (Split-Path $PSScriptRoot -Parent) -Parent) 'modules'),
```

Comment:
This is something used in other scripts as well. I don't have a strong preference, but I wonder what's the advantage of this default value compared to the fact that we are forcing the script to be in a specific location in order to retrieve the modules folder + we are hardcoding the modules folder name as well. 
This function is anyway called by the pipeline passing parameter values to it, so the default values are never used and risk to get stale overtime if we for instance ever restructure the utilities folder. Food for thoughts. Anyway, I wouldn't block the PR on that since the same is done extensively throughout PS functions.

Reply:
I actually ran the script most of the time locally. The docs update was an afterthought. The default value is by the way not enforcing the location of the script but merely makes it easier to use if the script is not moved. I don't really know what see any disadvantage here

Reply:
As mentioned, this is not blocking so let's discuss it offline. The main disadvantage I see is the risk of getting the default value stale. I'm sure you tested that locally in implementation phase. I'm not sure this would be often the case after merge.